bundles/org.openhab.transform.scale/src/main/java/org/openhab/transform/scale/internal/ScaleTransformationService.java

```diff
@@ -125,24 +147,29 @@ private String getScaleResult(Map<Range, String> data, String source, final BigD
                 final String value = properties.getProperty(entry);
                 final Matcher matcher = LIMITS_PATTERN.matcher(entry);
                 if (matcher.matches() && (matcher.groupCount() == 4)) {
-                    final boolean lowerInclusive = matcher.group(1).equals("]") ? false : true;
-                    final boolean upperInclusive = matcher.group(4).equals("[") ? false : true;
+                    final boolean lowerInclusive = matcher.group(1).equals("[");
+                    final boolean upperInclusive = matcher.group(4).equals("]");
 
                     final String lowLimit = matcher.group(2);
                     final String highLimit = matcher.group(3);
 
                     try {
-                        final BigDecimal lowValue = lowLimit.isEmpty() ? null : new BigDecimal(lowLimit);
-                        final BigDecimal highValue = highLimit.isEmpty() ? null : new BigDecimal(highLimit);
-                        final Range range = Range.range(lowValue, lowerInclusive, highValue, upperInclusive);
+                        final Range range = Range.range(lowLimit.isEmpty() ? null : new BigDecimal(lowLimit),
```

Comment:
That's fully correct code-wise, but hard to read. would you mind reverting that?